Sources/SwiftGenKit/Utils/YAML.swift

```diff
@@ -61,3 +61,50 @@ public enum YAML {
     }
   }
 }
+
+// Copied from /Source/SwiftLintFramework/Models/YamlParser.swift
+// of https://github.com/realm/SwiftLint/blob/d1dbc31aa9269364d6d7f43d2f99c82e12ceca6f
+private extension Constructor {
+  static func swiftgenContructor(env: [String: String]) -> Constructor {
+    return Constructor(customScalarMap(env: env))
+  }
+
+  static func customScalarMap(env: [String: String]) -> ScalarMap {
+    var map = defaultScalarMap
+    map[.str] = String.constructExpandingEnvVars(env: env)
+    map[.bool] = Bool.constructUsingOnlyTrueAndFalse
+
+    return map
+  }
+}
+
+private extension String {
+  static func constructExpandingEnvVars(env: [String: String]) -> (_ scalar: Node.Scalar) -> String? {
+    return { (scalar: Node.Scalar) -> String? in
+      scalar.string.expandingEnvVars(env: env)
+    }
+  }
+
+  func expandingEnvVars(env: [String: String]) -> String {
+    var result = self
+    for (key, value) in env {
+      result = result.replacingOccurrences(of: "${\(key)}", with: value)
+    }
+
+    return result
+  }
+}
+
+private extension Bool {
+  // swiftlint:disable:next discouraged_optional_boolean
+  static func constructUsingOnlyTrueAndFalse(from scalar: Node.Scalar) -> Bool? {
```

Comment:
I'd have to check from a computer to have the big picture and navigate in the codebase at ease, but didn't we have that logic for special handling of booleans in the YAML already? I remember we had to make sure `true` would be interpreted as Boolean but `"true"` as a string and thus already had something for that…

Reply:
Yeah, like I said in a previous comment, we don't need the custom boolean logic, we already solve that by enforcing all strings to have `"` (quotes).